### PR TITLE
Simplify doc build process by removing empty directories locally

### PR DESCRIPTION
**Pull request by kwankyu**
Now doc builder removes empty directories only in its own source directory.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by vbraun** [comment]
Bootstrap download of the confball does not work (note the missing version):
```
$ ./bootstrap -u http://sagepad.org/spkg/upstream/configure
[...]
Error: downloading configure-.tar.gz from http://sagepad.org/spkg/upstream/configure failed
```

**Comment by vbraun** [comment]
Needs work because it has #37858 merged in

**Comment by kwankyu** [comment]
This PR is now separate from #37858. Positive review?

**Comment by mkoeppe** [comment]
Has the code that removes empty directories been merged already?

**Comment by kwankyu** [comment]
Before this PR, empty directories in all docs (each of them is a sphinx project) are removed at the same time when references (BIBLIO in src/doc/Makefile, one particular sphinx project) is made. This approach works but violates , well, the principle of separation of concerns(?).

With this PR, empty directories in each doc are removed when that doc is made. So one sphinx project does not touch empty directories in governance of another sphinx project. This approach also works and looks better.

**Comment by mkoeppe** [comment]
Right, that's what I understood from the PR description, but in the diff I only see that `--no-prune-empty-dirs` is no longer used.

**Comment by kwankyu** [comment]
In the phase of building inventory, `--no-prune-empty-dirs` is not used, so that empty directories are removed. In the phase of building html files, `--no-prune-empty-dirs` is used (as before) so that no time is wasted to remove empty directories.

File: src/doc/Makefile [hunk 1]
```diff
@@ -63,7 +63,7 @@ doc-html-reference: doc-html-reference-sub
 doc-html-other: doc-html-reference
 	$(eval DOCS = $(shell sage --docbuild --all-documents all))
 	@if [ -z "$(DOCS)" ]; then echo "Error: 'sage --docbuild --all-documents' failed"; exit 1; fi
-	$(MAKE) SAGE_DOCBUILD_OPTS="$(SAGE_DOCBUILD_OPTS) --no-prune-empty-dirs" $(foreach doc, $(wordlist 2, 100, $(DOCS)), doc-html--$(subst /,-,$(doc)))
+	$(MAKE) $(foreach doc, $(wordlist 2, 100, $(DOCS)), doc-html--$(subst /,-,$(doc)))
```
Comment (mkoeppe) [inline src/doc/Makefile:66]: but this change here is for the "HTML" phase?

Comment (kwankyu) [reply src/doc/Makefile:66]: Above this part, it is for reference manual (that has inventory phase).

Here, it is for all other docs different from reference manual (that have only html phase).

Comment (kwankyu) [reply src/doc/Makefile:66]: Above this part, it is for sub docs of the reference manual (that has inventory phase).

Comment (mkoeppe) [reply src/doc/Makefile:66]: OK, thanks for the explanation.

**Comment by kwankyu** [comment]
Thanks!

**Pull request merged**